### PR TITLE
Feature/builder pickup

### PR DESCRIPTION
**Pull request by Raycoms**
# Changes proposed in this pull request:
- Let builder pickup items in the building after finishing it.
- If not reachable remove item from list after time.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java [hunk 1]
```diff
@@ -21,8 +21,10 @@
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.item.EntityArmorStand;
 import net.minecraft.entity.item.EntityItemFrame;
+import net.minecraft.entity.monster.EntityMob;
 import net.minecraft.init.Blocks;
 import net.minecraft.init.Items;
+import net.minecraft.item.ItemMonsterPlacer;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java]: MINOR Remove this unused import 'net.minecraft.item.ItemMonsterPlacer'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 49 issues
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 41 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractEntityAIStructure.java#L59](https://github.com/Minecolonies/minecolonies/blob/cd78601345029c867de3ae9b1cf7e82246946785/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L59): class "AbstractEntityAIStructure" has 37 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [AbstractEntityAIStructure.java#L777](https://github.com/Minecolonies/minecolonies/blob/cd78601345029c867de3ae9b1cf7e82246946785/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L777): Refactor this method to reduce its Cognitive Complexity from 32 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L49](https://github.com/Minecolonies/minecolonies/blob/cd78601345029c867de3ae9b1cf7e82246946785/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L49): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 31 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L186](https://github.com/Minecolonies/minecolonies/blob/cd78601345029c867de3ae9b1cf7e82246946785/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L186): The Cyclomatic Complexity of this method "requestMaterials" is 17 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L269](https://github.com/Minecolonies/minecolonies/blob/cd78601345029c867de3ae9b1cf7e82246946785/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L269): Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [Structure.java#L65](https://github.com/Minecolonies/minecolonies/blob/cd78601345029c867de3ae9b1cf7e82246946785/src/main/java/com/minecolonies/coremod/entity/ai/util/Structure.java#L65): Reduce this class from 47 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MINOR][MINOR] [blockout](https://github.com/Minecolonies/minecolonies/blob/cd78601345029c867de3ae9b1cf7e82246946785/src/main/java/com/minecolonies/blockout): Add a 'package-info.java' file to document the 'blockout' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [controls](https://github.com/Minecolonies/minecolonies/blob/cd78601345029c867de3ae9b1cf7e82246946785/src/main/java/com/minecolonies/blockout/controls): Add a 'package-info.java' file to document the 'controls' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [views](https://github.com/Minecolonies/minecolonies/blob/cd78601345029c867de3ae9b1cf7e82246946785/src/main/java/com/minecolonies/blockout/views): Add a 'package-info.java' file to document the 'views' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/cd78601345029c867de3ae9b1cf7e82246946785/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 48 issues
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 40 minor
* ![INFO][INFO] 2 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractEntityAIStructure.java#L59](https://github.com/Minecolonies/minecolonies/blob/a4bde7ebac2908ccfb14eb2206918f9a20e1ee23/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L59): class "AbstractEntityAIStructure" has 37 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [AbstractEntityAIStructure.java#L777](https://github.com/Minecolonies/minecolonies/blob/a4bde7ebac2908ccfb14eb2206918f9a20e1ee23/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L777): Refactor this method to reduce its Cognitive Complexity from 32 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L48](https://github.com/Minecolonies/minecolonies/blob/a4bde7ebac2908ccfb14eb2206918f9a20e1ee23/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L48): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 31 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L185](https://github.com/Minecolonies/minecolonies/blob/a4bde7ebac2908ccfb14eb2206918f9a20e1ee23/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L185): The Cyclomatic Complexity of this method "requestMaterials" is 17 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L268](https://github.com/Minecolonies/minecolonies/blob/a4bde7ebac2908ccfb14eb2206918f9a20e1ee23/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L268): Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [Structure.java#L65](https://github.com/Minecolonies/minecolonies/blob/a4bde7ebac2908ccfb14eb2206918f9a20e1ee23/src/main/java/com/minecolonies/coremod/entity/ai/util/Structure.java#L65): Reduce this class from 47 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MINOR][MINOR] [blockout](https://github.com/Minecolonies/minecolonies/blob/a4bde7ebac2908ccfb14eb2206918f9a20e1ee23/src/main/java/com/minecolonies/blockout): Add a 'package-info.java' file to document the 'blockout' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [controls](https://github.com/Minecolonies/minecolonies/blob/a4bde7ebac2908ccfb14eb2206918f9a20e1ee23/src/main/java/com/minecolonies/blockout/controls): Add a 'package-info.java' file to document the 'controls' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [views](https://github.com/Minecolonies/minecolonies/blob/a4bde7ebac2908ccfb14eb2206918f9a20e1ee23/src/main/java/com/minecolonies/blockout/views): Add a 'package-info.java' file to document the 'views' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/a4bde7ebac2908ccfb14eb2206918f9a20e1ee23/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by marvin-bitterlich** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/999/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 47 issues
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 39 minor
* ![INFO][INFO] 2 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractEntityAIStructure.java#L59](https://github.com/Minecolonies/minecolonies/blob/a4bde7ebac2908ccfb14eb2206918f9a20e1ee23/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L59): class "AbstractEntityAIStructure" has 37 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [AbstractEntityAIStructure.java#L777](https://github.com/Minecolonies/minecolonies/blob/a4bde7ebac2908ccfb14eb2206918f9a20e1ee23/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L777): Refactor this method to reduce its Cognitive Complexity from 32 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L48](https://github.com/Minecolonies/minecolonies/blob/a4bde7ebac2908ccfb14eb2206918f9a20e1ee23/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L48): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 31 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L185](https://github.com/Minecolonies/minecolonies/blob/a4bde7ebac2908ccfb14eb2206918f9a20e1ee23/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L185): The Cyclomatic Complexity of this method "requestMaterials" is 17 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L268](https://github.com/Minecolonies/minecolonies/blob/a4bde7ebac2908ccfb14eb2206918f9a20e1ee23/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L268): Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [Structure.java#L65](https://github.com/Minecolonies/minecolonies/blob/a4bde7ebac2908ccfb14eb2206918f9a20e1ee23/src/main/java/com/minecolonies/coremod/entity/ai/util/Structure.java#L65): Reduce this class from 47 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MINOR][MINOR] [blockout](https://github.com/Minecolonies/minecolonies/blob/a4bde7ebac2908ccfb14eb2206918f9a20e1ee23/src/main/java/com/minecolonies/blockout): Add a 'package-info.java' file to document the 'blockout' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [controls](https://github.com/Minecolonies/minecolonies/blob/a4bde7ebac2908ccfb14eb2206918f9a20e1ee23/src/main/java/com/minecolonies/blockout/controls): Add a 'package-info.java' file to document the 'controls' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [views](https://github.com/Minecolonies/minecolonies/blob/a4bde7ebac2908ccfb14eb2206918f9a20e1ee23/src/main/java/com/minecolonies/blockout/views): Add a 'package-info.java' file to document the 'views' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/a4bde7ebac2908ccfb14eb2206918f9a20e1ee23/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**